packages/core-data/README.md

```diff
@@ -764,6 +764,57 @@ In the above example, when `PageTitleDisplay` is rendered into an
 application, the page and the resolution details will be retrieved from
 the store state using `getEntityRecord()`, or resolved if missing.
 
+```js
+import { useState } from '@wordpress/data';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { TextControl } from '@wordpress/components';
+import { store as noticeStore } from '@wordpress/notices';
+import { useEntityRecord } from '@wordpress/core-data';
+
+function PageRenameForm( { id } ) {
+	const page = useEntityRecord( 'postType', 'page', id );
+	const [ title, setTitle ] = useState( () => page.record.title.rendered );
```

Comment:
One thing that still isn't clear to me. Why do we need `useState` here rather than using `const title = page.editedRecord.title` and `const setTitle = ( value ) => page.edit( { title: value } )`?

Reply:
`const setTitle = ( value ) => page.edit( { title: value } )` would make undo work character by character :(

Reply:
I'll go ahead and merge this PR, we can tweak the docs in a follow-up if needed.

Reply:
I approved the PR. I'm wondering how the undo/redo works with `setAttributes` from the block edit API. I don't see there an issue with storing all atomic operations when using `onChange` with `TextControl`.

Reply:
I figured out why it works in a special way for block attributes. There is this util that checks whether the changes get applied to the same block's attribute:

https://github.com/WordPress/gutenberg/blob/e2b293d2cfcd8c01c778780fa319a025d464b652/packages/block-editor/src/store/reducer.js#L154-L172

It's used by the higher-order reducer that ensures that only a single entry gets added to the under/redo state for subsequent changes to the same attribute:

https://github.com/WordPress/gutenberg/blob/e2b293d2cfcd8c01c778780fa319a025d464b652/packages/block-editor/src/store/reducer.js#L463-L517

Maybe we should figure out how to mirror the same behavior for entities to unify the experience and bring more value to using `editedRecord` and the `edit` method from this hook.